client/my-sites/plans-v2/upsell.tsx

```diff
@@ -1,41 +1,187 @@
+import {
+	durationToString,
+	durationToText,
+	getProductPrices,
+	getProductUpsell,
+	slugToSelectorProduct,
+} from './utils';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
 
 /**
  * Type dependencies
  */
-import type { UpsellPageProps } from './types';
+import type { SelectorProduct, UpsellPageProps } from './types';
+
+// There are several things that could go wrong depending on the URL
+// - The product slug could not exist.
+// - The product slug could be of a product that does not have another product to
+// upsell to – Jetpack Anti-Spam is an example.
+// - The product slug could have an upsell product but what if the user already has that
+// product?
+// - What if the user already owns the product associated with the product slug?
+
+interface Props {
+	currencyCode: string;
+	mainProduct: SelectorProduct;
+	selectedSiteSlug: string;
+	upsellProduct: SelectorProduct;
+}
+
+const UpsellComponent = ( {
+	currencyCode,
+	mainProduct,
+	selectedSiteSlug,
+	upsellProduct,
+}: Props ) => {
+	const translate = useTranslate();
+
+	const availableProducts = useSelector( ( state ) => getAvailableProductsList( state ) );
+
+	// Upsell prices should come from the API with the upsell products
+	const upsellProductPrices = getProductPrices( upsellProduct, availableProducts );
+	const savedAmount = 100;
+
+	const { shortName: mainProductName, productSlug } = mainProduct;
+	const { shortName: upsellProductName, productSlug: upsellProductSlug } = upsellProduct;
+
+	const goToCheckout = useCallback( () => {
+		page( `/checkout/${ selectedSiteSlug }` );
+	}, [ selectedSiteSlug ] );
+
+	const onPurchaseBothProducts = useCallback( () => {
+		addItems( [ jetpackProductItem( productSlug ), jetpackProductItem( upsellProductSlug ) ] );
+		goToCheckout();
+	}, [ goToCheckout, productSlug, upsellProductSlug ] );
+
+	const onPurchaseSingleProduct = useCallback( () => {
+		addItem( jetpackProductItem( productSlug ) );
+		goToCheckout();
+	}, [ goToCheckout, productSlug ] );
+
+	return (
+		<Main className="upsell">
+			<div className="upsell__header">
+				<FormattedHeader
+					headerText={ preventWidows(
+						translate( 'Would you like to add %s?', {
+							args: [ upsellProductName ],
+							comment: '%s is the name of a product such as Jetpack Scan or Jetpack Backup',
+						} )
+					) }
+					brandFont
+				/>
+				<div className="upsell__icons">
+					<ProductIcon className="upsell__product-icon" slug={ mainProduct.iconSlug } />
+					<Gridicon className="upsell__plus-icon" icon="plus-small" />
+					<ProductIcon className="upsell__product-icon" slug={ upsellProduct.iconSlug } />
+				</div>
+				<p className="upsell__subheader">
+					{ preventWidows(
+						translate( 'Bundle %(mainProduct)s with %(upsellProduct)s and save!', {
+							args: {
+								mainProduct: mainProductName,
+								upsellProduct: upsellProductName,
+							},
+							comment: '%s are abbreviated name of product such as Scan or Backup',
```

Comment:
Let's be specific as to what `%s` the comment is referring to, in case this string ever has something else added.

Reply:
Yeah, those are mistakes from a copy-paste.